### PR TITLE
use empty slice as largest/smallest key for compact_range

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
rocksdb_compact_range use nullptr as largest/smallest key.

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
@BusyJay @siddontang @ngaut PTAL

File: src/rocksdb.rs [hunk 1]
```diff
@@ -850,10 +850,20 @@ impl DB {
 
     pub fn compact_range(&self, start_key: &[u8], end_key: &[u8]) {
         unsafe {
+            let start = if start_key.is_empty() {
+                0 as *const u8
```
Comment (BusyJay) [inline src/rocksdb.rs]: Use `ptr::null()` instead.

File: src/rocksdb.rs [hunk 2]
```diff
@@ -850,10 +850,20 @@ impl DB {
 
     pub fn compact_range(&self, start_key: &[u8], end_key: &[u8]) {
         unsafe {
+            let start = if start_key.is_empty() {
```
Comment (BusyJay) [inline src/rocksdb.rs]: Seems more appropriate to use `Option`.

**Comment by BusyJay** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
@siddontang @ngaut PTAL

**Comment by ngaut** [comment]
LGTM

**Pull request merged**